hazelcast/src/main/java/com/hazelcast/internal/ascii/rest/HttpPostCommandProcessor.java

```diff
@@ -621,21 +627,58 @@ private static String exceptionResponse(Throwable throwable) {
         return response(ResponseType.FAIL, "message", throwable.getMessage());
     }
 
-    private static String response(ResponseType type, String... attributes) {
+    private static String response(ResponseType type, Object... attributes) {
         final StringBuilder builder = new StringBuilder("{");
         builder.append("\"status\":\"").append(type).append("\"");
         if (attributes.length > 0) {
             for (int i = 0; i < attributes.length; ) {
-                final String key = attributes[i++];
-                final String value = attributes[i++];
+                final String key = attributes[i++].toString();
+                final Object value = attributes[i++];
                 if (value != null) {
-                    builder.append(String.format(",\"%s\":\"%s\"", key, value));
+                    builder.append(String.format(",\"%s\":%s", key, jsonLiteral(value)));
                 }
             }
         }
         return builder.append("}").toString();
     }
 
+    private static String jsonLiteral(Object value) {
+        if (value instanceof String) {
+            return '"' + (String) value + '"';
+        } else if (value instanceof Collection) {
+            return "[" + join((Collection) value, ",") + "]";
+        } else {
+            throw new IllegalArgumentException("Unsupported ");
+        }
+    }
+
+    public static String join(Collection strings, final String separator) {
```

Comment:
javaDoc?

Reply:
Added the javadoc and simplified the method since it's always using the same separator and renamed it to reflect that it always produces JSON.